packages/perseus-core/src/utils/add-library-version-To-perseus-debug.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Adds the given perseus library version information to the __perseus_debug__
+ * object and ensures that the object is attached to `globalThis` (`window` in
+ * browser environments).
+ *
+ * This allows each library to provide runtime version information to assist in
+ * debugging in production environments.
+ */
+export const addLibraryVersionToPerseusDebug = (
+    libraryName: string,
+    libraryVersion: string,
+) => {
+    // If the library version is the default value, then we don't want to
+    // prefix it with a "v" to indicate that it is a version number.
+    let prefix = "v";
+    if (libraryVersion === "__lib_version__") {
+        prefix = "";
+    }
+    const s = `${prefix}${libraryVersion}`;
+
+    if (globalThis) {
+        globalThis.__perseus_debug__ = globalThis.__perseus_debug__ ?? {};
+
+        globalThis.__perseus_debug__[libraryName] = s;
```

Comment:
I think if we already have one in stored we should check to see if the version matches the old version - and if not we should turn this into an array. This will help you debug cases where we have multiple versions of the same library running side-by-side.

Reply:
I like these ideas. Thank-you. 

I've updated this function to do as suggested: first registration registers the key as the library name and the value as the version string. On subsequent calls for the same library, the entry is morphed into an array that includes a distinct list of versions registered. Added tests to validate all this.